omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,214 @@
+			String prefix = postBody.get("prefix").toString();
+			String suffix = postBody.get("suffix").toString();
+			Integer minLength = Integer.parseInt(postBody.get("minLength").toString());
+			Integer maxLength = Integer.parseInt(postBody.get("maxLength").toString());
+			
+			identifierSource.setIdentifierType(Context.getPatientService().getPatientIdentifierTypeByUuid(uuid));
+			identifierSource.setName(name);
+			identifierSource.setDescription(description);
+			identifierSource.setBaseCharacterSet(baseCharacterSet);
+			identifierSource.setFirstIdentifierBase(firstIdentifierBase);
+			identifierSource.setPrefix(prefix);
+			identifierSource.setSuffix(suffix);
+			identifierSource.setMinLength(minLength);
+			identifierSource.setMaxLength(maxLength);
+			
+			savedIdentifierSource =  save(identifierSource);	
+		}
+		
+		else if(postBody.get("sourceType").toString().equals("RemoteIdentifierSource")) {
+			
+			RemoteIdentifierSource identifierSource = new RemoteIdentifierSource();
+			
+			String uuid = postBody.get("identifierType").toString();
+			String name = postBody.get("name").toString();
+			String description = postBody.get("description").toString();
+			String username = postBody.get("username").toString();
+			String password = postBody.get("password").toString();
+			String url = postBody.get("url").toString();
+			
+			identifierSource.setIdentifierType(Context.getPatientService().getPatientIdentifierTypeByUuid(uuid));
+			identifierSource.setName(name);
+			identifierSource.setUrl(url);
+			identifierSource.setDescription(description);
+			identifierSource.setUser(username);
+			identifierSource.setPassword(password);
+			
+			savedIdentifierSource =  save(identifierSource);
+			
+		}
+		
+		else {
+			
+			IdentifierPool identifierSource = new IdentifierPool();
+			
+			String uuid = postBody.get("identifierType").toString();
+			String name = postBody.get("name").toString();
+			Integer batchSize = Integer.parseInt(postBody.get("batchSize").toString());
+			Integer minPoolSize = Integer.parseInt(postBody.get("minPoolSize").toString());
+			String sourceUuid = postBody.get("sourceUuid").toString();
+			Boolean sequential = Boolean.parseBoolean(postBody.get("sequential").toString());
+			Boolean refillWithScheduledTask = Boolean.parseBoolean(postBody.get("refillWithScheduledTask").toString());
+			
+			identifierSource.setIdentifierType(Context.getPatientService().getPatientIdentifierTypeByUuid(uuid));
+			identifierSource.setName(name);
+			identifierSource.setBatchSize(batchSize);
+			identifierSource.setMinPoolSize(minPoolSize);
+			identifierSource.setRefillWithScheduledTask(refillWithScheduledTask);
+			identifierSource.setSequential(sequential);
+			identifierSource.setSource(Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid));
+			
+			Set<PooledIdentifier> pooledIdentifiers = new LinkedHashSet<PooledIdentifier>();
+			if(sequential) {
```

Comment:
@dkayiwa please comment about this if statement.

what should be the value of "ABC0" => prefix or first identifier base or something else?